timeMachine.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+var timeMachine = ( function iife() {
+
+  var month = new Array( 12 ) ;
+  month[0] = 'January';
+  month[1] = 'February';
+  month[2] = 'March';
+  month[3] = 'April';
+  month[4] = 'May';
+  month[5] = 'June';
+  month[6] = 'July';
+  month[7] = 'August';
+  month[8] = 'September';
+  month[9] = 'October';
+  month[10] = 'November';
+  month[11] = 'December';
+
+  var weekday = new Array( 7 );
+  weekday [0] = 'Sunday';
+  weekday [1] = 'Monday';
+  weekday [2] = 'Tuesday';
+  weekday [3] = 'Wednesday';
+  weekday [4] = 'Thursday';
+  weekday [5] = 'Friday';
+  weekday [6] = 'Saturday';
+
+  var internalDate = null;
+
+  function dateInstantiate() {
+    if ( internalDate === null ) {
+      internalDate = new Date();
+
+      return internalDate;
+    }
+  }
+
+  function setDate( dateInput ) {
+    dateInstantiate();
+
+    if ( dateInput instanceof Date ) {
+      internalDate = dateInput;
```

Comment:
You don't want to store a reference to the date instance passed in. Instead, you want to create a new Date instance, passing in dateInput to the Date constructor when you do this:

`if (dateInput instanceof Date) { internalDate = new Date(dateInput); }`

The reason you don't want to store a reference to the date instance passed in is because it then allows code outside the module to modify the date you store internally. Because dateInput is a reference, modifying the original reference modifies your internalDate reference.